elfloader.c

```diff
@@ -91,8 +97,13 @@ int main(int argc, char **argv)
 
     entrypoint_t ep = (entrypoint_t) epaddr;
 
-    printf("Binary entrypoint is %x; invoking %p\n", ctx.ehdr.e_entry, ep);
+    printf("Binary entrypoint is %lx; invoking %p\n", ctx.ehdr.e_entry, ep);
```

Comment:
e_entry is an Elf_Addr, pedantically this should probably be PRIxPTR and a cast to (uintptr_t) to ensure the types match.

(%lx is likely to cause compiler warnings on 32-bit platforms where Elf_Addr is quite possibly int)